internal/procedures/server/start.go

```diff
@@ -47,8 +50,15 @@ func Start(initiator common.ServerInitiator) (err error) {
 		go startCourse(course)
 	}
 
-	// Cleanup any temp dirs.
-	defer util.RemoveRecordedTempDirs()
+	// Don't remove temp dirs during unit testing since they are needed after this function finishes.
+	if !config.UNIT_TESTING_MODE.Get() {
```

Comment:
I would like to avoid special cases for testing if we can
(that was a large chunk of our last chat).

Can this be handled with an argument instead (cleanupTempDirs?)?